apps/asap-server/test/handlers/research-output/index-handler.test.ts

```diff
@@ -20,36 +22,379 @@ describe('Research Output index handler', () => {
 
   afterEach(() => jest.clearAllMocks());
 
-  test('Should fetch the research-output and create a record in Algolia when no records exists yet', async () => {
+  test('Should fetch the research-output and create a record in Algolia when research-output is created', async () => {
     const researchOutputResponse = getResearchOutputResponse();
     researchOutputControllerMock.fetchById.mockResolvedValueOnce(
       researchOutputResponse,
     );
 
-    await indexHandler(getEvent());
+    await indexHandler(createEvent('ro-1234'));
 
     expect(algoliaIndexMock.saveObject).toHaveBeenCalledWith({
       ...researchOutputResponse,
       objectID: researchOutputResponse.id,
     });
   });
+
+  test('Should fetch the research-output and create a record in Algolia when research-output is updated', async () => {
+    const researchOutputResponse = getResearchOutputResponse();
+    researchOutputControllerMock.fetchById.mockResolvedValueOnce(
+      researchOutputResponse,
+    );
+
+    await indexHandler(updateEvent('ro-1234'));
+
+    expect(algoliaIndexMock.saveObject).toHaveBeenCalledWith({
+      ...researchOutputResponse,
+      objectID: researchOutputResponse.id,
+    });
+  });
+
+  test('Should remove the record Algolia when research-output is unpublished', async () => {
+    const event = unpublishedEvent('ro-1234');
+    await indexHandler(event);
+
+    expect(algoliaIndexMock.deleteObject).toHaveBeenCalledWith(
+      event.detail.payload.id,
+    );
+  });
+
+  test('Should remove the record Algolia when research-output is deleted', async () => {
+    const event = deleteEvent('ro-1234');
+    await indexHandler(event);
+
+    expect(algoliaIndexMock.deleteObject).toHaveBeenCalledWith(
+      event.detail.payload.id,
+    );
+  });
+
+  describe('Should process all events, and not relay on the order', () => {
```

Comment:
i think the point of these tests is that these events happen almost at the same time - we are testing race conditions and the fact that they should not rely on the event data (or the order like you said). i think its important that the description outlines that so it should be something along the lines of `Should process the events, handle race conditions and not rely on the order of the events`